### ISSUE TITLE
Network plugin type change doesn't discuss impact / not accurate

### ISSUE THREAD

**Issue by smarterclayton** [openshift/openshift-docs#1941]
https://docs.openshift.org/latest/install_config/configuring_sdn.html#migrating-between-sdn-plugins

Does not accurately reflect the process to migrate between plugins, especially impact to cluster.  User reported having to redeploy all pods in all nodes in order to make it effective.  If that's the case, we should have much deeper description of the impact.

@knobunc

**Comment by danwinship** [comment]
The procedure described in the docs is supposed to work. If people had to redeploy pods rather than just restarting openshift on each node, then that's a bug in openshift, not in the docs.

**Comment by smarterclayton** [comment]
@anthonyrisinger was the one who hit this, I believe - he believes the upgrade process did not work from the docs - can you comment here?

**Comment by exowaucka** [comment]
I work with Anthony.  We found that we needed to rebuild the pods.  A simple redeploy did not fix the lack of connectivity (redeploys would fail because they couldn't pull the image from the registry).

**Comment by exowaucka** [comment]
Update: I think the registry problems when redploying were caused by a misconfiguration: we gave the registry the wrong name for the pvc, so it didn't actually store anything in the pv.  This has been rectified.

**Comment by anthonyrisinger** [comment]
Even after fixing that things don't feel right... the bad rules resulting from https://github.com/openshift/openshift-sdn/pull/285 (erroneous namespace mismatches causing service lockout) are being restored after restarting `origin-node.service`:

```
I0426 22:43:46.868804   14232 proxier.go:938] Restoring iptables rules: *filter
```

Which just adds all the lockout rules back even though the subnet plugin is in play. I'm still convinced swapping plugins is broken or needs verification, *especially* after triggering the above bug (could be why we thought it didn't work initially).

**Comment by anthonyrisinger** [comment]
OK I think maybe the "no endpoints" thing is always added(?) and then removed later when the master announces endpoints? or something?

At any rate, if you guys really say it should Just Work, and your (proper) setups demonstrate this, then it probably does and we simply created a Bad Day for ourselves.

**Comment by mburke5678** [comment]
@smarterclayton @danwinship What is the status of this issue? Is there still an issue in the docs that needs to be addressed or was the misconfiguration exowaucka talked about the problem?

**Comment by anthonyrisinger** [comment]
I'm pretty confident the issue we encountered was self-inflicted. I don't
believe the network plugin switch had anything to do with the actual
problem at the time. We just scapegoated it before it became painfully
clear our registry wasn't properly configured for persistence. The symptoms
we saw looked like network resolution problems initially.

I've been through the related network plugin code quite a bit at this
point, and don't see any reason one can't swap them at will.

We have nothing new to add and the platform has ran swimmingly since. No
need to keep this ticket open on our account.

-- 

C Anthony
